batch/sql/estimated-current.sql

```diff
@@ -1480,6 +1553,56 @@ BEGIN
   SELECT 0 as rc, delta_cores_mcpu;
 END $$
 
+DROP PROCEDURE IF EXISTS mark_job_group_complete $$
+CREATE PROCEDURE mark_job_group_complete(
+  IN in_batch_id BIGINT,
+  IN in_job_group_id INT,
+  IN new_timestamp BIGINT
+)
+BEGIN
+  DECLARE cursor_job_group_id INT;
+  DECLARE done BOOLEAN DEFAULT FALSE;
+  DECLARE total_jobs_in_job_group INT;
+  DECLARE cur_n_completed INT;
+
+  DECLARE job_group_cursor CURSOR FOR
+  SELECT parent_id
+  FROM job_group_parents
+  WHERE batch_id = in_batch_id AND job_group_id = in_job_group_id
+  ORDER BY job_group_id ASC;
+
+  DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
```

Comment:
This is pretty heady stuff! Worth a link to https://dev.mysql.com/doc/refman/8.0/en/cursors.html

Reply:
For this I meant we need a comment with that link